frontend/src/container/MetricsApplication/MetricsPageQueries/MetricsPageQueriesFactory.ts

```diff
@@ -184,12 +184,140 @@ export const getQueryBuilderQuerieswithAdditionalItems = ({
 	],
 });
 
+export const getQueryBuilderQueriesTop = ({
+	metricName,
+	servicename,
+	tagFilterItems,
+	legend,
+	topLevelOperations,
+}: BuilderQueriesPropsTop): {
+	formulas: IMetricsBuilderFormula[];
+	queryBuilder: IMetricsBuilderQuery[];
+} => ({
+	formulas: [],
+	queryBuilder: [
+		{
+			aggregateOperator: 18,
+			disabled: false,
+			legend,
+			metricName,
+			name: 'A',
+			reduceTo: 1,
+			tagFilters: {
+				items: [
+					{
+						id: '',
+						key: 'service_name',
+						op: 'IN',
+						value: [`${servicename}`],
+					},
+					{
+						id: '',
+						key: 'operation',
+						op: 'MATCH',
+						value: topLevelOperations,
+					},
+					...tagFilterItems,
+				],
+				op: 'AND',
+			},
+		},
+	],
+});
+export const getQueryBuilderQueriesTopWithFormula = ({
+	servicename,
+	legend,
+	legendFormula,
+	disabled,
+	tagFilterItems,
+	topLevelOperations,
+	metricNameA,
+	metricNameB,
+	groupBy,
+	expression,
+	additionalItems,
+}: BuilderQueriesTopWithFormulaProps): {
+	formulas: IMetricsBuilderFormula[];
+	queryBuilder: IMetricsBuilderQuery[];
+} => ({
+	formulas: [
+		{
+			disabled: false,
+			expression,
+			name: 'F1',
+			legend: legendFormula,
+		},
+	],
+	queryBuilder: [
+		{
+			aggregateOperator: 18,
+			disabled,
+			groupBy,
+			legend,
+			metricName: metricNameA,
+			name: 'A',
+			reduceTo: 1,
+			tagFilters: {
+				items: [
+					{
+						id: '',
+						key: 'service_name',
+						op: 'IN',
+						value: [`${servicename}`],
+					},
+					{
+						id: '',
+						key: 'operation',
+						op: 'MATCH',
+						value: topLevelOperations,
+					},
+					additionalItems,
+					...tagFilterItems,
+				],
+
+				op: 'AND',
+			},
+		},
+		{
+			aggregateOperator: 18,
+			disabled,
+			groupBy,
+			legend,
+			metricName: metricNameB,
+			name: 'B',
+			reduceTo: 1,
+			tagFilters: {
+				items: [
+					{
+						id: '',
+						key: 'service_name',
+						op: 'IN',
+						value: [`${servicename}`],
+					},
+					{
+						id: '',
+						key: 'operation',
+						op: 'MATCH',
+						value: topLevelOperations,
+					},
+					...tagFilterItems,
+				],
+				op: 'AND',
+			},
```

Comment:
This is entirely common for both queries. What's the point of repeating same?

Reply:
So you think I could just declare something called items and then pass it to both the queries? @srikanthccv

Reply:
Yes, you could name something like `commonFilterItems` or something.